src/modules/blocks/event-links/template.js

```diff
@@ -89,48 +89,69 @@ const renderButtons = ( props ) => (
 	</div>
 );
 
-const renderControls = ({
-	hasGoogleCalendar,
-	hasiCal,
+const renderControls = ( {
+	attributes,
 	isSelected,
 	toggleIcalLabel,
 	toggleGoogleCalendar,
-}) => (
-	isSelected && (
-		<InspectorControls key="inspector">
-			<PanelBody title={ __( 'Share Settings', 'the-events-calendar' ) }>
-				<ToggleControl
-					label={ __( 'Google Calendar', 'the-events-calendar' ) }
-					checked={ hasGoogleCalendar }
-					onChange={ toggleGoogleCalendar }
-				/>
-				<ToggleControl
-					label={ __( 'iCal', 'the-events-calendar' ) }
-					checked={ hasiCal }
-					onChange={ toggleIcalLabel }
-				/>
-			</PanelBody>
-		</InspectorControls>
-	)
-);
+} ) => {
+	const { hasGoogleCalendar, hasiCal } = attributes;
 
-const EventLinks = ( props ) => (
-	[
-		renderButtons( props ),
-		renderControls( props ),
-	]
-);
+	return (
+		isSelected && (
+			<InspectorControls key="inspector">
+				<PanelBody title={ __( 'Share Settings', 'the-events-calendar' ) }>
+					<ToggleControl
+						label={ __( 'Google Calendar', 'the-events-calendar' ) }
+						checked={ hasGoogleCalendar }
+						onChange={ toggleGoogleCalendar }
+					/>
+					<ToggleControl
+						label={ __( 'iCal', 'the-events-calendar' ) }
+						checked={ hasiCal }
+						onChange={ toggleIcalLabel }
+					/>
+				</PanelBody>
+			</InspectorControls>
+		)
+	);
+};
+
+const EventLinks = ( props ) => {
+	const { setAttributes } = props;
+
+	const setiCalLabel = e => setAttributes( { iCalLabel: e.target.value } );
+	const setGoogleCalendarLabel = e => setAttributes( { googleCalendarLabel: e.target.value } );
+	const toggleIcalLabel = value => setAttributes( { hasiCal: value } );
+	const toggleGoogleCalendar = value => setAttributes( { hasGoogleCalendar: value } );
+
+	const combinedProps = {
+		...props,
+		setiCalLabel,
+		setGoogleCalendarLabel,
+		toggleIcalLabel,
+		toggleGoogleCalendar,
+	};
+
+	return [
+		renderButtons( combinedProps ),
+		renderControls( combinedProps ),
+	];
+};
 
 EventLinks.propTypes = {
-	hasGoogleCalendar: PropTypes.bool,
-	hasiCal: PropTypes.bool,
+	attributes: PropTypes.object,
 	isSelected: PropTypes.bool,
-	googleCalendarLabel: PropTypes.string,
-	iCalLabel: PropTypes.string,
-	setiCalLabel: PropTypes.func,
-	setGoogleCalendarLabel: PropTypes.func,
-	toggleIcalLabel: PropTypes.func,
-	toggleGoogleCalendar: PropTypes.func,
+	setAttributes: PropTypes.func,
+
+	// hasGoogleCalendar: PropTypes.bool,
```

Comment:
We're leaving this commented? Maybe a `@todo` to come back later? Or notes for context?

Reply:
Nope, that's something I missed after doing some testing. Nice catch.